server/src/main/java/org/apache/iotdb/db/utils/QueryDataSetUtils.java

```diff
@@ -67,13 +67,15 @@ public static TSQueryDataSet convertQueryDataSetByFetchSize(
       if (queryDataSet.hasNext()) {
         RowRecord rowRecord = queryDataSet.next();
         // filter rows whose columns are null according to the rule
-        if ((queryDataSet.isWithoutAllNull() && rowRecord.isAllNull())
-            || (queryDataSet.isWithoutAnyNull() && rowRecord.hasNullField())) {
-          // if the current RowRecord doesn't satisfy, we should also decrease AlreadyReturnedRowNum
+        if (queryDataSet.withoutNullFilter(
+            rowRecord, queryDataSet.isWithoutAllNull(), queryDataSet.isWithoutAnyNull())) {
```

Comment:
`queryDataSet.isWithoutAllNull(), queryDataSet.isWithoutAnyNull()` is not needed.

Reply:
fix